src/clue.js

```diff
@@ -2,25 +2,152 @@
+         image: "https://www.radiotimes.com/uploads/images/Original/111967.jpg",
+         color: "red"
+         
+     },
+     {  
+         firstName: "Eleanor",
+         lastName: "Peacock",
+         occupation: "Socialité",
+         age: 36,
+         description: "She is from a wealthy family and uses her status and money to earn popularity",
+         image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+         color: "blue"
+         
+     }, 
+    {   
+      
+         firstName: "Jack",
+         lastName: "Mustard",
+         occupation: "Retired Football player",
+         age: 62,
+         description: "He is a former football player who tries to get by on his former glory",
+         image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+         color: "yellow"
+         
+     }
+     
+ ];
+ 
+ // Rooms Array
+ 
+ const roomsArray = [
+     {name: "Dining Room"},
+     {name: "Conservatory"},
+     {name: "Kitchen"},
+     {name: "Study"},
+     {name: "Library"},
+     {name: "Billiard Room"},
+     {name: "Lounge"},
+     {name: "Ballroom"},
+     {name: "Hall"},
+     {name: "Spa"},
+     {name: "Living Room"},
+     {name: "Observatory"},
+     {name: "Theater"},
+     {name: "Guest House"},
+     {name: "Patio"}
+ ];
+ 
+ // Weapons Array
+ 
+ const weaponsArray = [
+     {
+     name: "rope", weight: 10},
+     {name: "knife",  weight: 8},
+     {name: "candlestick",  weight: 2},
+     {name: "dumbbell",  weight: 30},
+     {name: "poison",  weight: 2},
+     {name: "axe",  weight: 15},
+     {name: "bat",  weight: 13},
+     {name: "trophy",  weight: 25},
+     {name: "pistol",  weight: 20}
+ 
+ ];
+ 
+ 
+ // ITERATION 2
+  
+ function selectRandom(array) {
+   if (array.length == 0) {
+     return undefined
+   }
+   if (array.length == 1) {
+     return array[0]
+   }
+   let randomPosition = Math.floor(Math.random(array) * array.length) ;
+   let randomWord = array[randomPosition]
+   //console.log(randomPosition);
+   //console.log(randomWord);
+   return randomWord 
+ }
+ /*console.log(selectRandom(roomsArray));
+ console.log(selectRandom(weaponsArray));
+ console.log(selectRandom(suspectsArray));
+ */
+ 
+ 
+ function pickMystery() {
+     let suspectRan = selectRandom(suspectsArray);
+     let weaponRan = selectRandom(weaponsArray);
+     let roomRan = selectRandom(roomsArray);
+     
+     let mistery = {
+         suspect: suspectRan,
+         weapon: weaponRan,
+         room: roomRan,
+     } 
+     console.log(mistery);
+     return mistery
+     
+ }
+ 
+ 
+ // ITERATION 3
+ 
+ function revealMystery() {
+    console.log(`${mistery.suspect.firstname}  ${mistery.suspect.lastName} killed Mr. Boddy using ${mistery.weapon} in the ${mistery.room}.`)
+     return `${mistery.suspect.firstname}  ${mistery.suspect.lastName} killed Mr. Boddy using ${mistery.weapon} in the ${mistery.room}.`
```

Comment:
killed Mr. Boddy using ${mistery.weapon} in the ${mistery.room}, nos falta el .name en ambos:
killed Mr. Boddy using the ${data.weapon.name} in the ${data.room.name}